python-sdk/actionml/__init__.py

```diff
@@ -0,0 +1,634 @@
+
+    def async_set_item(self, iid, properties=None, event_time=None):
+        """
+        Set properties of an item.
+        Wrapper of async_create function, setting event to "$set" and entity_type
+        to "item".
+        """
+        if properties is None:
+            properties = {}
+        return self.async_create(
+            event="$set",
+            entity_type="item",
+            entity_id=iid,
+            properties=properties,
+            event_time=event_time)
+
+    def set_item(self, iid, properties=None, event_time=None):
+        """Set properties of an item."""
+        if properties is None:
+            properties = {}
+        return self.async_set_item(iid, properties, event_time).get_response()
+
+    def async_unset_item(self, iid, properties=None, event_time=None):
+        """
+        Unset properties of an item.
+        Wrapper of async_create function, setting event to "$unset" and entity_type
+        to "item".
+        """
+        if properties is None:
+            properties = {}
+        return self.async_create(
+            event="$unset",
+            entity_type="item",
+            entity_id=iid,
+            properties=properties,
+            event_time=event_time)
+
+    def unset_item(self, iid, properties=None, event_time=None):
+        """Unset properties of an item."""
+        if properties is None:
+            properties = {}
+        return self.async_unset_item(iid, properties, event_time).get_response()
+
+    def async_delete_item(self, iid, event_time=None):
+        """
+        Delete an item.
+        Wrapper of async_create function, setting event to "$delete" and entity_type
+        to "item".
+        """
+        return self.async_create(
+            event="$delete",
+            entity_type="item",
+            entity_id=iid,
+            event_time=event_time)
+
+    def delete_item(self, iid, event_time=None):
+        """Delete an item."""
+        return self.async_delete_item(iid, event_time).get_response()
+
+    def async_record_user_action_on_item(self, action, uid, iid, properties=None, event_time=None):
+        """
+        Create a user-to-item action.
+        Wrapper of async_create function, setting entity_type to "user" and
+        target_entity_type to "item".
+        """
+        if properties is None:
+            properties = {}
+        return self.async_create(
+            event=action,
+            entity_type="user",
+            entity_id=uid,
+            target_entity_type="item",
+            target_entity_id=iid,
+            properties=properties,
+            event_time=event_time)
+
+    def record_user_action_on_item(self, action, uid, iid, properties=None, event_time=None):
+        """Create a user-to-item action."""
+        if properties is None:
+            properties = {}
+        return self.async_record_user_action_on_item(
+            action, uid, iid, properties, event_time).get_response()
+
```

Comment:
Most of these should be replaced with event construction like the Java SDK to keep this code decoupled from specific Templates